src/Bonsai.ML.PointProcessDecoder/CreatePointProcessModel.cs

```diff
@@ -0,0 +1,448 @@
+        }
+        set
+        {
+            maxCovariateRange = value;
+        }
+    }
+
+    private long[] stepsCovariateRange = [50];
+    /// <summary>
+    /// Gets or sets the number of steps evaluated in the covariate range. Must be the same length as the number of covariate dimensions.
+    /// </summary>
+    [Category("2. Covariate Parameters")]
+    [Description("The number of steps evaluated in the covariate range. Must be the same length as the number of covariate dimensions.")]
+    [TypeConverter(typeof(UnidimensionalArrayConverter))]
+    public long[] Steps
+    {
+        get
+        {
+            return stepsCovariateRange;
+        }
+        set
+        {
+            stepsCovariateRange = value;
+        }
+    }
+
+    private double[] covariateBandwidth = [1];
+    /// <summary>
+    /// Gets or sets the kernel bandwidth used to estimate the probability density over the covariate dimensions. Must be the same length as the covariate dimensions.
+    /// </summary>
+    [Category("2. Covariate Parameters")]
+    [Description("The kernel bandwidth used to estimate the probability density over the covariate dimensions. Must be the same length as the covariate dimensions.")]
+    [TypeConverter(typeof(UnidimensionalArrayConverter))]
+    public double[] Bandwidth
+    {
+        get
+        {
+            return covariateBandwidth;
+        }
+        set
+        {
+            covariateBandwidth = value;
+        }
+    }
+
+    private EncoderType encoderType = EncoderType.SortedSpikes;
+    /// <summary>
+    /// Gets or sets the type of encoder used.
+    /// </summary>
+    [Category("3. Encoder Parameters")]
+    [Description("The type of encoder used.")]
+    public EncoderType EncoderType
+    {
+        get
+        {
+            return encoderType;
+        }
+        set
+        {
+            encoderType = value;
+        }
+    }
+
+    private int? kernelLimit = null;
+    /// <summary>
+    /// Gets or sets the maximum number of kernels maintained in memory for each probability density estimation made by the encoder. In the case of sorted spikes, there is an estimate for the full covariate distribution and an estimate for each unit. In the case of clusterless marks, there is an estimate for the full covariate distribution, and 2 estimates for each mark channel (1 for the overall distribution of marks and 1 for the joint distribution of covariates and marks).
+    /// </summary>
+    [Category("3. Encoder Parameters")]
+    [Description("The maximum number of kernels maintained in memory for each probability density estimation made by the encoder. In the case of sorted spikes, there is an estimate for the full covariate distribution and an estimate for each unit. In the case of clusterless marks, there is an estimate for the full covariate distribution, and 2 estimates for each mark channel (1 for the overall distribution of marks and 1 for the joint distribution of covariates and marks).")]
```

Comment:
I would avoid duplicating long remarks across both XML and description attribute strings. Ideally we would add the long description in a `<remarks>` element in the base documentation, and leave it to the future to incorporate these in the editor. Currently it is very hard to read long strings in the property grid text box anyway so it is best to rely on "View help".

Reply:
@ncguilbeault I am happy to merge this without this being addressed but potentially it might be worth discussing how to deal with these for the future in a separate issue. The description box in the property grid is not really well designed for long documentation strings so I tend to prefer keeping the description strings quite short.